src/palace/manager/api/admin/controller/work_editor.py

```diff
@@ -367,45 +368,58 @@ def edit(self, identifier_type, identifier):
 
         return Response("", 200)
 
-    def suppress(self, identifier_type, identifier):
-        """Suppress the license pool associated with a book."""
-        self.require_librarian(flask.request.library)
+    def suppress(
+        self, identifier_type: str, identifier: str
+    ) -> Response | ProblemDetail:
+        """Suppress a book at the level of a library."""
+
+        library = flask.request.library  # type: ignore[attr-defined]
+
+        if library is None:
+            return self.no_library_response()
 
-        # Turn source + identifier into a LicensePool
-        pools = self.load_licensepools(
-            flask.request.library, identifier_type, identifier
+        self.require_library_manager(library)
+
+        work = self.load_work(
+            library=library, identifier_type=identifier_type, identifier=identifier
         )
-        if isinstance(pools, ProblemDetail):
+
+        if isinstance(work, ProblemDetail):
             # Something went wrong.
-            return pools
+            return work
+
+        work.suppressed_for.append(library)
+        self.log.info(
+            f"Suppressed {identifier_type}/{identifier} (work id: {work.id}) for library {library.short_name}."
+        )
 
-        # Assume that the Work is being suppressed from the catalog, and
-        # not just the LicensePool.
-        # TODO: Suppress individual LicensePools when it's not that deep.
-        for pool in pools:
-            pool.suppressed = True
         return Response("", 200)
 
-    def unsuppress(self, identifier_type, identifier):
-        """Unsuppress all license pools associated with a book.
+    def unsuppress(
+        self, identifier_type: str, identifier: str
+    ) -> Response | ProblemDetail:
+        """Remove a book suppression from a book at the level of a library"""
 
-        TODO: This will need to be revisited when we distinguish
-        between complaints about a work and complaints about a
-        LicensePoool.
-        """
-        self.require_librarian(flask.request.library)
+        library = flask.request.library  # type: ignore[attr-defined]
+
+        if library is None:
+            return self.no_library_response()
```

Comment:
Same here re: `LIBRARY_NOT_FOUND` problem detail.

Reply:
I actually think that this should return a 404, but fine to leave this as-is here and I can update in the in-progress follow-up PR.

Reply:
Fixed.